tools/bgpreader.c

```diff
@@ -596,13 +644,26 @@ int main(int argc, char *argv[])
     goto err;
   }
 
+#ifdef WITH_RPKI
+  if(rpki_input != NULL && rpki_input->rpki_active){
```

Comment:
maybe change to `goto err;` in the `if` statement, so that there is only one place where you destroy the related objects.

Reply:
Yes, it would work and save a line of code, but I think it's a little bit confusing to jump to ```err:``` if the run is successful, don't you?

Reply:
Agreed. I thought this was for error handling. My mistake.